backend/icpy/agent/tools/atlascloud/v2v_sound_tool.py

```
@@ -0,0 +1,565 @@
+            
+            # If path is already absolute, use it as-is, otherwise join with workspace root
+            if os.path.isabs(clean_video):
+                full_path = Path(clean_video)
+            else:
+                full_path = Path(workspace_root) / clean_video
+        else:
+            # Remote context
+            full_path = Path(clean_video)
+        
+        if not full_path.exists():
+            raise ValueError(f"Video file not found: {video} (resolved to {full_path})")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)